packages/bindings/src/mocks/mock_keeper.rs

```diff
@@ -1,103 +1,103 @@
+//! when performing integration tests.
+
 #![cfg(not(tarpaulin_include))]
+use crate::mocks::mock_queriers::MockDesmosQuerier;
 use crate::msg::DesmosMsg;
-use crate::query::DesmosQuery;
-use anyhow::Result as AnyResult;
-use cosmwasm_std::{Addr, Api, Binary, BlockInfo, Empty, Event, Querier, Storage};
-use cw_multi_test::{AppResponse, CosmosRouter, Module};
-use std::convert::TryFrom;
-
 #[cfg(feature = "posts")]
-use crate::posts::{mocks::mock_posts_query_response, msg::PostsMsg};
+use crate::posts::msg::PostsMsg;
 #[cfg(feature = "profiles")]
-use crate::profiles::{mocks::mock_profiles_query_response, msg::ProfilesMsg};
+use crate::profiles::msg::ProfilesMsg;
+use crate::query::DesmosQuery;
 #[cfg(feature = "reactions")]
-use crate::reactions::{mocks::mock_reactions_query_response, msg::ReactionsMsg};
+use crate::reactions::msg::ReactionsMsg;
 #[cfg(feature = "relationships")]
-use crate::relationships::{mocks::mock_relationships_query_response, msg::RelationshipsMsg};
+use crate::relationships::msg::RelationshipsMsg;
 #[cfg(feature = "reports")]
-use crate::reports::{mocks::mock_reports_query_response, models::ReportTarget, msg::ReportsMsg};
+use crate::reports::{models::ReportTarget, msg::ReportsMsg};
 #[cfg(feature = "subspaces")]
-use crate::subspaces::{mocks::mock_subspaces_query_response, msg::SubspacesMsg};
+use crate::subspaces::msg::SubspacesMsg;
+use anyhow::Result as AnyResult;
+use cosmwasm_std::{
+    Addr, Api, Binary, BlockInfo, ContractResult, Empty, Event, Querier, QueryRequest, Storage,
+};
+use cw_multi_test::{AppResponse, CosmosRouter, Module};
+use std::convert::TryFrom;
 
 /// Represents the implementation of [`Module`](cw_multi_test::Module) for handling the desmos execution and query messages.
 #[derive(Default)]
-pub struct DesmosKeeper {}
+pub struct DesmosKeeper {
+    /// Querier used to handle the query requests.
+    pub querier: MockDesmosQuerier,
+}
 
 impl DesmosKeeper {
     /// Returns a new [DesmosKeeper].
     pub fn new() -> Self {
-        DesmosKeeper {}
+        DesmosKeeper {
+            querier: MockDesmosQuerier::new(&[]),
+        }
+    }
+
+    /// Returns a new [DesmosKeeper] with a custom instance of [MockDesmosQuerier].
+    pub fn with_custom_querier(querier: MockDesmosQuerier) -> Self {
+        DesmosKeeper { querier }
     }
 
-    /// Handles [`ProfilesMsg`](crate::profiles::ProfilesMsg) then returns the response with proper [events](https://github.com/desmos-labs/desmos/blob/master/x/profiles/spec/05-events.md).
+    /// Handles [`ProfilesMsg`](crate::profiles::msg::ProfilesMsg) then returns the response with proper [events](https://github.com/desmos-labs/desmos/blob/master/x/profiles/spec/05-events.md).
     #[cfg(feature = "profiles")]
-    fn handle_profiles_msg(&self, block: &BlockInfo, msg: ProfilesMsg) -> AnyResult<AppResponse> {
+    pub fn handle_profiles_msg(block: &BlockInfo, msg: ProfilesMsg) -> AnyResult<AppResponse> {
```

Comment:
Should we have a `MockMsgHandler` to allow devs to build custom msg handler as well? What do you think?

Reply:
I think that this is not required. 
The only think that can be usefull is if a developer perform actions with the attributes retuned from the execution of one of the our modules but i can't think at any actual use caso for that.
Maybe we can open a low priority issue to keep track of that if some one needs this in the future.

Reply:
Agree with you, I think it is enough with current one. Let's push it as a low priority feature.